vcs/fake.go

```diff
@@ -0,0 +1,53 @@
+package vcs
+
+import (
+	"crypto/sha1"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+func init() {
+	Register(newFake, "fake")
+}
+
+type FakeDriver struct{}
+
+func newFake(b []byte) (Driver, error) {
+	return &FakeDriver{}, nil
+}
+
+func (g *FakeDriver) HeadRev(dir string) (string, error) {
+	hsh := sha1.New()
+	cmd := exec.Command(
+		"tar",
+		"cf", "-",
+		"./",
+	)
+	cmd.Dir = dir
+	cmd.Stdout = hsh
```

Comment:
Clever! I really like that this doesn't leave tar files in the filesystem and doesn't (or shouldn't) be a problem with large inputs.